R/app.R

```diff
@@ -51,28 +41,22 @@ plot_callback <- function(input, output, session){
 router <- make_router(
   route("/", main_page, main_callback),
   route("plot", plot_page, plot_callback),
-  page_404 = p404
+  page_404 = page404(message404 = "Nie ma takiej strony")
 )
 
```

Comment:
ptal, at line 34: `x <- faithful[, 2]` can be set outside server function. As a result x will be calculated once, and each user will not have to recalculate it on new session.